config/initializers/yabeda.rb

```diff
@@ -2,6 +2,6 @@
 
 require 'three_scale/metrics/yabeda'
 
-Rails.application.config.to_prepare do
+Rails.application.config.after_initialize do
   ThreeScale::Metrics::Yabeda.install!
```

Comment:
Do you think this will be enough? If `ThreeScale::Metrics::Yabeda` is reloaded, then I expect that `ThreeScale::Metrics::Yabeda::controller_handlers` will be reset. So the `process_action.action_controller` notification will stop working.

Idk if this is important as I'm not sure we care about prometheus in dev mode.

Reply:
Hm... not sure :upside_down_face: 
What do you suggest?

Reply:
You already said it - to move it back to `/lib`... in case there is no way to eliminate state. What is this `ThreeScale::Metrics::Yabeda::controller_handlers` used for? I can't see them set anywhere. Can we remove them?

Reply:
Good catch @akostadinov. If `ThreeScale::Metrics::Yabeda` is realoadable, the `@controller_handlers` value is reset after every change in the code, because the old class is replaced by a new one. However, that's not happening right now because of the `require` in the initializer. Removing the `require` makes the `Yabeda` class reloadable and thus the `@controller_handlers` starts getting reset.

Reply:
So to have it reloadable we need to:

1. remove the `require`
2. bring it back inside the `to_prepare`

Correct?

Reply:
I mean, in this way it will reinitialize on every reload and the `@controller_handlers` will be set again. Although I didn't see anything setting them, but I mean if there was something setting them.

Reply:
You're right @akostadinov This is the extension mechanism by [yabeda-rails](https://github.com/yabeda-rb/yabeda-rails#hooks). We don't need it, so I'm removing. Thanks!

Reply:
> I think that we still need to have it in after_initialize, see https://github.com/3scale/porta/pull/3593#discussion_r1364132705

Yes